pkg/provision/kubernetes.go

```diff
@@ -0,0 +1,311 @@
+package provision
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/zos/pkg"
+	"github.com/threefoldtech/zos/pkg/stubs"
+)
+
+// Kubernetes reservation data
+type Kubernetes struct {
+	// Size of the vm, this defines the amount of vCpu, memory, and the disk size
+	Size uint8 `json:"size"`
+
+	// NetworkID of the network namepsace in which to run the VM. The network
+	// must be provisioned previously.
+	NetworkID pkg.NetID `json:"network_id"`
```

Comment:
Can't we reuse the same type used in container reservation to configure the network ?
So we are consistent between both ?

Reply:
They don't match one to one except for the NetworkID, but containers 'Network' struct also has List of IPs while a machine has only one.

Reply:
Actually for container we only assign 1 address. Maybe we should then modify the container network config to also only accept one IP.